controller/class.tapestry-controller.php

```diff
@@ -43,56 +43,76 @@ class TapestryController
         'INVALID_CHILD_NODE' => [
             'MESSAGE'   => 'Node is not a child of the tapestry',
             'STATUS'    => ['status' => 400]
+        ],
+        'INVALID_NEW_TAPESTRY' => [
+            'MESSAGE'   => 'Settings do not exist when creating a new Tapestry',
+            'STATUS'    => ['status' => 400]
+        ],
+        'INVALID_NEW_LINK' => [
+            'MESSAGE'   => 'New Tapestry link is invalid',
+            'STATUS'    => ['status' => 400]
+        ],
+        'NODES_EXIST_IN_NEW_TAPESTRY' => [
+            'MESSAGE'   => 'Nodes exist when creating a new Tapestry',
+            'STATUS'    => ['status' => 400]
+        ],
+        'GROUPS_EXIST_IN_NEW_TAPESTRY' => [
+            'MESSAGE'   => 'Groups exist when creating a new Tapestry',
+            'STATUS'    => ['status' => 400]
+        ],
+        'LINKS_EXIST_IN_NEW_TAPESTRY' => [
+            'MESSAGE'   => 'Links exist when creating a new Tapestry',
+            'STATUS'    => ['status' => 400]
+        ],
+        'POST_ID_ALREADY_SET' => [
+            'MESSAGE'   => 'PostID is set when creating a new Tapestry',
+            'STATUS'    => ['status' => 500]
         ]
     ];
     private $postId;
 
     /**
      * Constructor
      */
-    public function __construct($postId = 0)
+    public function __construct($postId = null)
     {
-        if ($postId && !$this->_isValidTapestry($postId)) {
-            return $this->_throwsError('INVALID_POST_ID');
+        if ($postId && $this->_isValidTapestry($postId)) {
+            $this->postId = (int)$postId;
+        } else {
+            $this->postId = null;
         }
-        $this->postId = (int)$postId;
     }
 
     /**
-     * Update Tapestry nodes first then
-     * Update the existing Tapestry if the postId is provided
-     * Otherwise, a new Tapestry will be created
+     * Add A Tapestry
      * 
      * @param   Object  $tapestry   Tapestry
      * 
      * @return  Object  $tapestry
      */
-    public function updateTapestry($tapestry)
+    public function addTapestry($tapestry)
     {
-        if (!$this->postId) {
-            $this->postId = $this->_updatePost($tapestry, 'tapestry');
+        if ($this->postId) {
+            return $this->_throwsError('POST_ID_ALREADY_SET');
         }
-        if (!isset($tapestry->nodes)) {
-            $tapestry->nodes = [];
+        if (empty($tapestry->settings)) {
+            return $this->_throwsError('INVALID_NEW_TAPESTRY');
```

Comment:
Again, I suggest you change this error index to be more descriptive like `'SETTINGS_MISSING_IN_NEW_TAPESTRY'` or something